account_invoice_policy_order_subtotal/models/sale_order_line.py

```diff
@@ -0,0 +1,116 @@
+        subtot_policy_lns = self._get_policy_order_subtotal_lines()
+        for line in subtot_policy_lns:
+            line.qty_to_invoice = line.product_uom_qty - line.qty_invoiced
+        super(SaleOrderLine, self - subtot_policy_lns)._get_to_invoice_qty()
+
+    @api.depends("invoice_lines.move_id.state", "invoice_lines.quantity")
+    def _get_invoice_qty(self):
+        subtot_policy_lns = self._get_policy_order_subtotal_lines()
+        for line in subtot_policy_lns:
+            line.qty_invoiced = (
+                line.product_uom_qty * line.amount_invoiced / line.price_subtotal
+            )
+        super(SaleOrderLine, self - subtot_policy_lns)._get_invoice_qty()
+
+    @api.depends("product_id")
+    def _compute_qty_delivered_method(self):
+        subtot_policy_lns = self._get_policy_order_subtotal_lines()
+        for line in subtot_policy_lns:
+            line.qty_delivered_method = False
+        super(SaleOrderLine, self - subtot_policy_lns)._compute_qty_delivered_method()
+
+    def _prepare_invoice_line(self, **optional_values):
+        if (
+            self.product_id.type == "consu"
+            and self.product_id.invoice_policy == "order_subtotal"
+        ):
+            # We set the quantity to ordered quantity and
+            # let the user decide how much he wants to invoice
+            optional_values["quantity"] = self.product_uom_qty
+            optional_values["price_unit"] = 0
+        return super()._prepare_invoice_line(**optional_values)
+
+    @api.depends(
+        "state",
+        "product_uom_qty",
+        "qty_delivered",
+        "qty_to_invoice",
+        "qty_invoiced",
+        "amount_invoiced",
+    )
+    def _compute_invoice_status(self):
+        precision = self.env["decimal.precision"].precision_get("Product Price")
+        subtot_policy_lns = self._get_policy_order_subtotal_lines()
+        for line in subtot_policy_lns:
+            if line.state not in ("sale", "done"):
+                line.invoice_status = "no"
+            elif (
+                float_compare(
+                    line.price_subtotal,
+                    line.amount_invoiced,
+                    precision_digits=precision,
+                )
+                > 0
+            ):
+                line.invoice_status = "to invoice"
+            elif (
+                float_compare(
+                    line.price_subtotal,
+                    line.amount_invoiced,
+                    precision_digits=precision,
+                )
+                <= 0
+            ):
+                line.invoice_status = "invoiced"
+            else:
+                line.invoice_status = "no"
+        super(SaleOrderLine, self - subtot_policy_lns)._compute_invoice_status()
+
+    amount_invoiced = fields.Float(
+        compute="_compute_amount_invoiced",
+        string="Invoiced Amount",
+        store=True,
+        readonly=True,
+        compute_sudo=True,
+        digits="Product Unit of Measure",
```

Comment:
I don't think we want to use UOM precision if we're managing amount there. I would expect to use the currency precision in that case.